pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +119,272 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Check if the MasterUserRecord already exists
+	mur := &toolchainv1alpha1.MasterUserRecord{}
+	// We can use the same NamespacedName from the request as the MasterUserRecord will have the same name as the
+	// UserSignup (i.e. the user's username)
+	err = r.client.Get(context.TODO(), request.NamespacedName, mur)
+	if err != nil {
+		// We generally EXPECT the MasterUserRecord to not be found here, so we only deal with other error types
+		if !errors.IsNotFound(err) {
+			// Error reading the MasterUserRecord, requeue the request
+			return reconcile.Result{}, err
+		}
+	} else {
+		// If we successfully found an existing MasterUserRecord then our work here is done, set the status
+		// to Complete and return
+		reqLogger.Info("MasterUserRecord exists, setting status to Complete")
+		statusError := r.setStatusComplete(instance, "")
+		if statusError != nil {
+			reqLogger.Error(statusError, "Error updating UserSignup Status to Complete", "Name", instance.Name)
+			return reconcile.Result{}, statusError
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// Check the user approval policy.
+	userApprovalPolicy, err := r.ReadUserApprovalPolicyConfig(request.Namespace)
+	if err != nil {
```

Comment:
For the consistency sake you might want to implement `wrapErrorWithStatusUpdate()` function to log the error and update the status in this controller. We use that approach in all our controllers.
Here are a few examples: 

https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccount/useraccount_controller.go#L152-L154

https://github.com/codeready-toolchain/host-operator/pull/23/files#diff-a7167111790d962cde50e8d4ded9aba5R156

Reply:
I've added this function and used it in a couple of places where appropriate, but in other cases I've used another new helper function which I added in order to clean up the code.  I'm still not happy with the way we do status updates, it still seems "clunky".  I'd like to come up with some more elegant way to manage the resource status if possible, so I'll spend some time thinking on this.